Download page template

slawekjaranowski opened the pull request.

slawekjaranowski committed:
Download page template

slawekjaranowski commented:
@kwin try to prepare template for download page ....

Questions:
 - should template be in parent pom project ... maybe in maven-apache-resources but it is for ASF, so maybe new project for Maven project templates ...

slawekjaranowski commented:
and using in plugin https://github.com/apache/maven-changes-plugin/pull/93

kwin commented on maven-shared-doc-resource/src/main/resources/markdown/download.md.vm:
```diff
@@ -0,0 +1,54 @@
+
+|  | Link                                                                                                                                                   |Checksum|Signature|
+|---|--------------------------------------------------------------------------------------------------------------------------------------------------------|---|---|
+${project.name} ${project.version} \(Source zip\)| [${downloadArtifactId}\-${project.version}\-source\-release\.zip](\[preferred\]/maven/${downloadDirectory}/${downloadArtifactId}-${project.version}-source-release.zip) |[${downloadArtifactId}\-${project.version}\-source\-release\.zip\.sha512](https://www.apache.org/dist/maven/${downloadDirectory}/${downloadArtifactId}-${project.version}-source-re ...
```
the placeholder `[preferred]` is actually replaced by a URL ending with slash (despite incorrect documentation at https://infra.apache.org/release-download-pages.html#custom), so it must not be followed by a slash.

kwin replied:
Let us use the new download location for checksum and signature: downloads.a.o instead of www.a.o/dist.

kwin commented on maven-shared-doc-resource/src/main/resources/resources/download.cgi:
we don't need this as this is globally setup: https://github.com/apache/infrastructure-p6/blob/1bd0e53353181cf46865b5ec8f5e4fa16541ce7e/modules/closer_cgi/files/closer-cgi.conf#L6C1-L6C12

kwin replied:
if we keep it locally we should point it to the correct script and also add a comment referencing https://github.com/apache/infrastructure-p6/tree/production/modules/closer_cgi/files and https://infra.apache.org/release-download-pages.html#custom

slawekjaranowski replied:
ok, first try with: https://github.com/apache/maven-site/pull/589

kwin commented on maven-shared-doc-resource/pom.xml:
```diff
@@ -0,0 +1,51 @@
+    <version>44-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>maven-shared-doc-resource</artifactId>
```
I would rather name this maven-shared-site-resource as the files in it are pretty maven-site specific. `doc` seems to generic for these files.

slawekjaranowski replied:
still is a question - should such template be placed in this project or some other?

kwin commented on maven-shared-doc-resource/src/main/resources/markdown/download.md.vm:
```diff
@@ -0,0 +1,54 @@
+under the License.
+*#
+#set( $h = "#" )
+#set( $downloadDirectory = $project.properties.downloadDirectory )
```
we need some documentation of the properties used by this template.

slawekjaranowski replied:
we we make decision where we should put it we will add a documentation

kwin commented on pom.xml:
```diff
@@ -1230,6 +1232,25 @@ under the License.
+              <goals>
+                <goal>process</goal>
+              </goals>
+              <phase>none</phase>
```
I would rather bind this to `pre-site`

slawekjaranowski replied:
due to inheritance we can put here pre-site

kwin commented on docs/pom.xml:
```diff
@@ -70,8 +70,27 @@ under the License.
+          <execution>
+            <id>site-download</id>
+            <phase>pre-site</phase>
+            <inherited>false</inherited>
```
I don't think anyone inherits from the site-pom anyways....

slawekjaranowski replied:
`docs/pom.xml` it is a parent pom for other docs site - without it it will be executed in all child

kwin replied:
Sorry, wasn't aware. Then obviously this is necessary.

kwin commented on docs/pom.xml:
```diff
@@ -70,8 +70,27 @@ under the License.
+        <executions>
+          <execution>
+            <id>site-download</id>
+            <phase>pre-site</phase>
```
instead of putting this config into each pom leveraging the templates, I would rather set https://maven.apache.org/plugins/maven-remote-resources-plugin/process-mojo.html#skip based on the values of `downloadDirectory`/`downloadArtifact` in the main parent pom.

slawekjaranowski replied:
in the most of we have a download page in project root pom, properties will also available in childs ...

kwin replied:
right, one possible way is to set the properties only for the current pom, but not if used as parent (https://github.com/apache/maven-apache-parent/blob/061e19af986a914df4d37b14e7e53a079d9033c5/pom.xml#L486)

slawekjaranowski replied:
I think it will be more complicated ... we will need a mark file in all project

kwin commented on maven-shared-doc-resource/src/main/resources/markdown/download.md.vm:
```diff
@@ -0,0 +1,54 @@
+
+In order to guard against corrupted downloads/installations, it is highly recommended to
+[verify the signature](https://www.apache.org/dev/release-signing#verifying-signature)
+of the release bundles against the public [KEYS](https://www.apache.org/dist/maven/KEYS) used by the Apache Maven
```
This link should point to downloads.a.o as well

kwin commented:
Maybe the approach in https://felix.apache.org/documentation/downloads.html (https://github.com/apache/felix-antora-site/edit/main/modules/ROOT/pages/downloads.adoc) works better. There the link is only resolved once the user clicks on download. No need for executing lua scripts (with potential performance penalty) for just rendering the download page itself.

slawekjaranowski commented:
> Maybe the approach in https://felix.apache.org/documentation/downloads.html (https://github.com/apache/felix-antora-site/edit/main/modules/ROOT/pages/downloads.adoc) works better. There the link is only resolved once the user clicks on download. No need for executing lua scripts (with potential performance penalty) for just rendering the download page itself. WDYT @slawekjaranowski ?

Looks ok, we will not a special steps to render page - simply we be static.

slawekjaranowski commented:
@kwin I hop now template looks better

I will move it to https://github.com/apache/maven-apache-resources where we have other resources bundles

kwin commented on pom.xml:
```diff
@@ -914,6 +914,7 @@ under the License.
     <module>maven-skins</module>
     <module>doxia-tools</module>
     <module>docs</module>
+    <module>maven-shared-doc-resource</module>
```
I will try to work on MSITE-911 in the near future, so this doesn't need to be maintained outside the parent project.

slawekjaranowski replied:
My approach to put it in external repo are:
- we can use it in ASF parent
- maybe other ASF project try to use it - eg commons use own template generated by deprecated Maven/Ant plugin https://github.com/apache/commons-build-plugin - @garydgregory WDYT

garydgregory replied:
Hello @slawekjaranowski 

The Commons Build and Release plugins are not deprecated and generates a lot more than just one page. I use them for all Commons releases.

What I'm more interested in getting rid of is the commons-skin plug-in but I don't know if it's possible to fold it in commons-parent directly.

It would be nice to merge the Build and Release plugin but I never took the time to do it. 

Ideally, I'd prefer to have all three current plug-ins in commons-parent, so I have 1 framework-y thing to deal with instead of 4.

slawekjaranowski replied:
I know that  Build and Release plugins are not deprecated but use deprecated ant mojo

slachiewicz commented:
Maybe instead of download page for each component in each project we could simply link to one "big" page on Maven site - with links to all latest source code downloads?

slawekjaranowski commented:
> Maybe instead of download page for each component in each project we could simply link to one "big" page on Maven site - with links to all latest source code downloads?

template in one place look also ok for me

slachiewicz committed:
Align the download template with the page it replaces

The template rendered a version-numbered "#Apache_Maven_Parent_POM_50-SNAPSHOT"
anchor and no "#Files". Every component already on the refreshed page publishes
"#Files", and a version-numbered anchor changes on each release, so keep the
heading literal.

The dlcdn host is deliberate (MNGSITE-505): the mirror network was retired, so
do not restore closer.lua or the [preferred] placeholder.

Verified: rendered docs/ site against the xdoc page from apache/maven-parent#593
  -> anchors, download hrefs and visible text all identical.

Generated-by: Claude Opus 5
